web/pandas/pdeps/0010-required-pyarrow-dependency.md

```diff
@@ -0,0 +1,68 @@
+  PyArrow version that has been released for at least 2 years, and the minimum PyArrow version will be
+  maintained for every minor version in the major version series.
+
+## Background
+
+PyArrow is an optional dependency of pandas that provides a wide range of supplemental features to pandas:
+
+- Since pandas 0.21.0, PyArrow provided I/O reading functionality for Parquet
+- Since pandas 1.2.0, pandas integrated PyArrow into the `ExtensionArray` interface to provide an optional string data type backed by PyArrow
+- Since pandas 1.4.0, PyArrow provided I/0 reading functionality for CSV
+- Since pandas 1.5.0, pandas provided an `ArrowExtensionArray` and `ArrowDtype` to support all PyArrow data types within the `ExtensionArray` interface
+- Since pandas 2.0.0, all I/O readers have the option to return PyArrow-backed data types, and many methods now utilize PyArrow compute functions to
+accelerate PyArrow-backed data in pandas, notibly string and datetime types.
+
+As of pandas 2.0, one can feasibly utilize PyArrow as an alternative data representation to NumPy with advantages such as:
+
+1. Consistent ``NA`` support for all data types
+2. Broader support of data types such as ``decimal``, ``date`` and nested types
+
+## Motivation
+
+While all the functionality described in the previous paragraph is currently optional, PyArrow has significant integration into many areas
+of pandas. With our roadmap noting that pandas strives for better Apache Arrow interoperability [^1] and many projects [^2], within or beyond the Python ecosystem, adopting or interacting with the Arrow format, making PyArrow a required dependency provides an additional signal of confidence in the Arrow
+ecosystem to pandas users.
+
+Additionally, requiring PyArrow would simplify the related development within pandas and potentially improve NumPy functionality that would be better suited
+by PyArrow including:
+
+- Avoiding runtime checking if PyArrow is available to perform PyArrow object inference during constructor or indexing operations
+  - Currently, there are 17 `import_optional_dependency("pyarrow")` checks throughout the pandas code base
+- Avoiding NumPy object data types more by default for analogous types that have native PyArrow support such as string, decimal, binary, and nested types
+
+## Drawbacks
+
+Including PyArrow would naturally increase the installation size of pandas. For example, installing pandas and PyArrow using pip from wheels, numpy and pandas
+are about `70MB`, and PyArrow is around `120MB`. An increase of installation size would have negative impliciation using pandas in space-constrained development
+or deployment environments such as AWS Lambda.
+
+Additionally, if a user is installing pandas in an environment where wheels are not available and needs to build from source, the user will need to build Arrow C++ and related dependencies. These environments include
```

Comment:
I think it would be good to more explicitly say that you need to do this (installing Arrow C++) manually and this is possible through `pip install pyarrow` 
(there are other python packages that also have C/C++ code but that do that build automatically (if you have the dependencies such as a compiler) when installing from source)

Reply:
thanks for providing details here - is this (`pip install pyarrow`) much of a hurdle for these cases where pandas wheels aren't available?

Reply:
Whoops, sorry there was a "not" missing in "_not_ possible through `pip install pyarrow`. Corrected now. 

But on the actual question how much of a hurdle this is: I would say, try it out yourself :) That's the best way to get an idea of how difficult it is, otherwise you can only take (or not) my words in saying that: yes, this is a huge hurdle. Installing pyarrow and Arrow C++ from source is far from trivial.

Reply:
thanks - from the installation instructions, it certainly seems tricky https://arrow.apache.org/docs/developers/cpp/building.html#